src/data/list/rotate.lean

```diff
@@ -376,6 +395,120 @@ begin
   exact ⟨λ ⟨n, hn, h⟩, ⟨n, nat.lt_succ_of_le hn, h⟩, λ ⟨n, hn, h⟩, ⟨n, nat.le_of_lt_succ hn, h⟩⟩
 end
 
+/-- List of all cyclic permutations of `l`.
+The list will always contain `length l` elements. This implies that under certain conditions,
+there are duplicates in `cyclic_permutations l`.
+The `n`th entry is equal to `l.rotate (n + 1)`, proven in `nth_le_cyclic_permutations`.
+The proof that every cyclic permutant of `l` is in the list is `mem_cyclic_permutations_iff`.
+
+     cyclic_permutations [1, 2, 3, 2, 4] =
+       [[1, 2, 3, 2, 4], [2, 3, 2, 4, 1], [3, 2, 4, 1, 2],
+        [2, 4, 1, 2, 3], [4, 1, 2, 3, 2]] -/
+def cyclic_permutations : list α → list (list α)
+| []       := [[]]
```

Comment:
Doesn't this directly contradict "The list will always contain `length l` elements."?

Reply:
Docstring clarified.